plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakeMetadata.java

```diff
@@ -486,7 +492,19 @@ public ConnectorTableMetadata getTableMetadata(ConnectorSession session, Connect
                 properties.buildOrThrow(),
                 Optional.ofNullable(tableHandle.getMetadataEntry().getDescription()),
                 constraints.stream()
-                        .map(SparkExpressionParser::toTrinoExpression)
+                        .map(constraint -> {
+                            try {
+                                return toTrinoExpression(constraint);
+                            }
+                            catch (TrinoException e) {
+                                if (e.getErrorCode().equals(DELTA_LAKE_UNSUPPORTED_EXPRESSION.toErrorCode()) && suppressExpressionParserFailure) {
+                                    LOG.debug("Failed to convert Spark check constraints to Trino expressions: %s", constraint);
+                                    return null;
```

Comment:
You can return `CAST(fail('Failed to convert Delta check constraints to Trino expression') AS boolean` as the constraint expression.
It wouldn't affect reads, and would fail same way for writes.
It would keep the code simpler.

Reply:
Thanks for your suggestion. Changed to use `fail` function.